File: src/main/java/com/dozycoffee/wms/warehouse/domain/valueobject/Capacity.java
package com.dozycoffee.wms.warehouse.domain.valueobject;

import com.dozycoffee.wms.warehouse.domain.exception.InvalidCapacityException;

public record Capacity(int value) {

    public Capacity {
        if (value < 0) {
            throw new InvalidCapacityException();
        }
    }
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/domain/valueobject/Capacity.kt
package com.dozycoffee.wms.warehouse.domain.valueobject

import com.dozycoffee.wms.warehouse.domain.exception.InvalidCapacityException

data class Capacity(val value: Int) {
    init {
        if (value < 0) {
            throw InvalidCapacityException()
        }
    }
}
